File: OOPS/circleconstrutor-project/src/com/circle/Circle.java
package com.circle;

import com.border.BorderStyle;

public class Circle {
	private float radius;
	private BorderStyle border;
	private static final double PI =3.14;
	
	
	public BorderStyle getBorder() {
		return border;
	}

	public Circle(float radius,BorderStyle border)
	{
		this.radius=radius;
		this.border=border;
	}
	
	public Circle(float radius)
	{
		this.radius=radius;
		border=BorderStyle.SINGLE_LINE;
	}
	
	public double calculateArea()
	{
		double area;
		area=PI*radius*radius;
		return area;
	}
	
	public double calculatePerimeter()
	{
		double perimeter;
		perimeter=2*PI*radius;
		return perimeter;
	}
	
	
}

File: CSharpOOPS/circle-consructor/circle-consructor/circle/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace circle_consructor.circle
{
    class Circle
    {
        private float _radius;
        private BorderStyle _border;
        private const float PI= 3.14f; 
        public enum BorderStyle
        {
            DOTTED, SINGLE_LINE, DOUBLE_LINE
        }

        public Circle(float radius, BorderStyle border)
        {
            _radius = radius;
            _border = border;
        }

        public float Radius
        {
            get
            {
                return _radius;
            }
        }

        public BorderStyle Border
        {
            get
            {
                return _border;
            }
        }
        public double calculateArea()
        {
            double area;
            area = PI * _radius * _radius;
            return area;
        }

        public double calculatePerimeter()
        {
            double perimeter;
            perimeter = 2 * PI * _radius;
            return perimeter;
        }
    }
}
